src/code.ts

```diff
@@ -77,13 +77,13 @@ export async function loadFile<Exports extends object = any>(
 }
 
 export async function writeFile(
-  node: { ast: ASTNode } | ASTNode,
+  node: { $ast: ASTNode } | ASTNode,
   filename?: string,
   options?: ParseOptions
 ): Promise<void> {
-  const ast = "ast" in node ? node.ast : node;
+  const ast = "$ast" in node ? node.$ast : node;
   const { code, map } = generateCode(ast, options);
-  filename = filename || (ast as any).name || "output.js";
+  filename = filename || "output.js";
```

Comment:
I kinda forgot why we have this optional. Maybe instead we could just make it mandatory.  A implicit file name feels weird.

Reply:
Updated!